var/spack/repos/builtin/packages/qgis/package.py

```diff
@@ -0,0 +1,214 @@
+    variant('qtwebkit',         default=False,  description='Enable QtWebkit Support')
+    variant('quick',            default=False,  description='Build QGIS Quick library')
+    variant('qwtpolar',         default=False,  description='Build QwtPolar')
+    variant('server',           default=False,  description='Build QGIS server')
+    variant('staged_plugins',   default=True,   description='Stage-install core Python plugins to run from build directory')
+    variant('thread_local',     default=True,   description='Use std::thread_local')
+    variant('txt2tags',         default=False,  description='Generate PDF for txt2tags documentation')
+
+    # Ref. for dependencies:
+    # http://htmlpreview.github.io/?https://raw.github.com/qgis/QGIS/master/doc/INSTALL.html
+    # https://github.com/qgis/QGIS/blob/master/INSTALL
+    depends_on('qt+dbus')
+    depends_on('proj@4.4.0:')
+    depends_on('geos@3.4.0:')
+    depends_on('sqlite@3.0.0: +column_metadata')
+    depends_on('libspatialite@4.2.0:')
+    depends_on('libspatialindex')
+    depends_on('gdal@2.1.0: +python', type=('build', 'link', 'run'))
+    depends_on('qwt@5:')
+    depends_on('qwtpolar')
+    depends_on('expat@1.95:')
+    depends_on('qca@2.2.1')
+    depends_on('py-pyqt4 +qsci', when='@2')
+    depends_on('py-pyqt5@5.3: +qsci', when='@3')
+    depends_on('qscintilla')
+    depends_on('qjson')
+    depends_on('py-requests', type=('build', 'run'))  # TODO: is build dependency necessary?
+    depends_on('py-psycopg2', type=('build', 'run'))  # TODO: is build dependency necessary?
+    depends_on('qtkeychain@0.5:', when='@3:')
+    depends_on('libzip')
+    depends_on('exiv2')
+    depends_on('python@3.0.0:', type=('build', 'run'), when='@3')
+    depends_on('python@2.7:2.8', type=('build', 'run'), when='@2')
+
+    # Runtime python dependencies, not mentioned in install instructions
+    depends_on('py-pyyaml', type='run')
+    depends_on('py-owslib', type='run')
+    depends_on('py-jinja2', type='run')
+    depends_on('py-pygments', type='run')
+
+    # optionals
+    depends_on('postgresql@8:', when='+postgresql')  # for PostGIS support
+    depends_on('gsl', when='+georeferencer')  # for georeferencer
+    depends_on('grass@7.0.0', type=('build', 'link', 'run'), when='+grass7')  # for georeferencer
+
+    # the below dependencies are shown in cmake config
+    depends_on('hdf5')
+    depends_on('netcdf-c')
+
+    # build
+    depends_on('cmake@3.0.0:', type='build')
+    depends_on('flex@2.5.6:', type='build')
+    depends_on('bison@2.4:', type='build')
+    depends_on('pkg-config', type='build')
+
+    # Conflicts for newer versions
+    conflicts('proj@:4.9.2', when='@3.8.2:')
+
+    # v3.8.1, Qt >= 5.9.0 is required
+    conflicts('qt@:5.8.99', when='@3.8.1:')
+    conflicts('qt@5.13.0:', when='@3.8.1')
+
+    # conflicts for qgis@2, qt@4, python@2
+    conflicts('qtkeychain@0.6.0:', when='^qt@4')
+    conflicts('qt@5:', when='@2')
```

Comment:
I think a lot of these conflicts could be reworked to be `depends_on` statements. Currently, `conflicts` doesn't influence the concretizer, it just causes it to fail.